futures-util/src/async_await/select.rs

```diff
@@ -14,58 +13,122 @@
 /// ```
 /// #![feature(pin, async_await, await_macro, futures_api)]
 /// # futures::executor::block_on(async {
-/// use futures::{select, future};
+/// use futures::future::{self, FutureExt};
+/// use futures::select;
 /// let mut a = future::ready(4);
-/// let mut b: future::Empty<()> = future::empty();
+/// let mut b = future::empty::<()>();
 ///
 /// let res = select! {
-///     a => a + 1,
-///     b => 0,
+///     future(a as a) => a + 1,
+///     future(b as _) => 0,
 /// };
 /// assert_eq!(res, 5);
 /// # });
 /// ```
+///
+/// In addition to `future(...)` matchers, `select!` accepts an `all complete`
+/// branch which can be used to match on the case where all the futures passed
+/// to select have already completed.
+///
+/// ```
+/// #![feature(pin, async_await, await_macro, futures_api)]
+/// # futures::executor::block_on(async {
+/// use futures::future::{self, FutureExt};
+/// use futures::select;
+/// let mut a = future::ready(4);
+/// let mut b = future::ready(6);
+/// let mut total = 0;
+///
+/// loop {
+///     select! {
+///         future(a as a) => total += a,
+///         future(b as b) => total += b,
+///         all complete => break,
```

Comment:
I find `all complete` a little confusing, it makes me think of a branch where all are _currently_ complete and ready to handle, not that they all had completed and been consumed in the past. I wonder if there’s a similarly succinct way to say this that implies that better.

Reply:
Yeah I puzzled on this a bit, but I think we're going to want to change this in the future anyways as we add cases for `next(...)` etc.